plugins/gssapi.c

```diff
@@ -1093,6 +1398,31 @@ gssapi_server_mech_authneg(context_t *text,
 	goto cleanup;
     }
 
+#ifdef GSS_USE_CCACHE_STORE
+    if (GSS_C_NO_CREDENTIAL != text->client_creds) {
+        const char* ccache = NULL;
+        const sasl_utils_t *utils = text->utils;
+        unsigned int rl;
+        utils->getopt(utils->getopt_context, "GSSAPI", "ccache_store", (const char**)&ccache, &rl);
+        if (NULL == ccache) {
+            text->ccname = NULL;
+        }
+        else {
+            text->ccname = store_client_credentials(text, &(text->client_creds),
+                                                    name_token.value, ccache);
```

Comment:
There is no guarantee that the value of name_token is zero terminated (in fact I know for a fact there are cases where it is not).
This is a bug throughout this function apparently and should be independently fixed (separate commit?)

But given you later want to use the gss name to resolve users, you may want to change the code to pass down name_token  (maybe change to use a better name?) whole here so you can later call gss_localname() on it.

Reply:
Right.. Yeah, it's just passed through two levels, so maybe that's a more logical design. I'll take a looksie. It is actually used, in the lower functions where a non-null-terminated string would be bad.

@simo5 a shame it's like that. Brings unnecessary boilerplate into the implementations. Feels like a bad design choice in whatever library implementation. Anyhow; I totally agree. I'll fix.

Reply:
@simo5 I don't feel i have (paid) time to refactor the whole thing (but I will make an effort on the new code I contributed), there are several strdup() and other things operating on only the .value element of the gss_buffer_desc elsewhere in gssapi_server_mech_authneg(). I could write some issues on various code-lines I've seen with similar use as I replicated tho.